akka-projection-core/src/main/scala/akka/projection/HandlerRecoveryStrategy.scala

```diff
@@ -77,12 +68,13 @@ object HandlerRecoveryStrategy {
    * INTERNAL API: placed here instead of the `internal` package because of sealed trait
    */
   @InternalApi private[akka] object Internal {
-    case object Fail extends HandlerRecoveryStrategy
-    case object Skip extends HandlerRecoveryStrategy
-    final case class RetryAndFail(retries: Int, delay: FiniteDuration) extends HandlerRecoveryStrategy {
+    case object Fail extends StrictRecoveryStrategy with RetryRecoveryStrategy
+    case object Skip extends StrictRecoveryStrategy with RetryRecoveryStrategy
```

Comment:
Good catch.  I'm not sure what you mean by a compile time test.  Would you use reflection to inspect types?

Reply:
No, no reflection. Compile time, but there are some limitation to this kind of test. We can only detect that we can call the API that we planned to make available, not the APIs that we don't want to make available. 

In the case here, we would have seen that although we had a `withRecoveryStrategy` in the `atMostOnce` we couldn't have called it because we didn't have access to a `StrictRecoveryStrategy`.

Another possible test, is to assign to a typed val after each method in the fluent API. That will detect when we by accident return the wrong type. Which was the case here: https://github.com/akka/akka-projection/pull/176#discussion_r431086931. We started with `atMostOnce` and we could end with an `atLeastOnce`.